src/screen.c

```diff
@@ -20,32 +18,34 @@ void screen_init()
 
 void screen_clear()
 {
+  u16 attrib = (0 << 4) | (0 & 0x0F);
+  volatile u16 * where;
+
   for (u16 x = 0; x < VGA_WIDTH; x++)
   {
     for (u16 y = 0; y < VGA_HEIGHT; y++)
     {
-      display_buffer[screen_position_to_index(x, y)] = '\0' | (7 << 8);
+      
+     where = (volatile u16 *)0xB8000 + (y * VGA_WIDTH + x) ;
+     *where = '\0' | (attrib << 8);
     }
   }
 }
 
 void screen_set_cursor_position(i8 x, i8 y)
 {
-  u16 index = screen_position_to_index(x, y + 1);
+  u16 index = (u16)(y * VGA_WIDTH + x);
```

Comment:
Use `screen_position_to_index` instead of manually calculating the index.

Reply:
screen_position_to_index is unnecessary as a function it's only used twice